telemetry/otlp_exporter.go

```diff
@@ -0,0 +1,162 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	otmetric "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+)
+
+const meterName = "cosmos-sdk-otlp-exporter"
+
+func StartOtlpExporter(cfg Config) {
+	ctx := context.Background()
+
+	exporter, err := otlpmetricgrpc.New(ctx,
+		otlpmetricgrpc.WithEndpoint(cfg.OtlpCollectorGrpcAddr),
+		otlpmetricgrpc.WithInsecure(),
+	)
+	if err != nil {
+		log.Fatalf("OTLP exporter setup failed: %v", err)
+	}
+
+	res, _ := resource.New(ctx, resource.WithAttributes(
+		semconv.ServiceName(cfg.OtlpServiceName),
+	))
+
+	meterProvider := metric.NewMeterProvider(
+		metric.WithReader(metric.NewPeriodicReader(exporter,
+			metric.WithInterval(cfg.OtlpPushInterval))),
+		metric.WithResource(res),
+	)
+	otel.SetMeterProvider(meterProvider)
+	meter := otel.Meter(meterName)
+
+	gauges := make(map[string]otmetric.Float64Gauge)
+	histograms := make(map[string]otmetric.Float64Histogram)
+
+	go func() {
```

Comment:
how does this behave when shutting a node down?  Do we want any kind of graceful shutdown here?